internal/conf/reposource/jvm_packages.go

```
@@ -0,0 +1,86 @@
+package reposource
+
+import (
+	"fmt"
+	"net/url"
+	"sort"
+	"strings"
+
+	"github.com/Masterminds/semver"
+
+	"github.com/sourcegraph/sourcegraph/internal/api"
+)
+
+type Module struct {
+	GroupId    string
+	ArtifactId string
+}
+
+func (m Module) MatchesDependencyString(dependency string) bool {
+	return strings.HasPrefix(dependency, fmt.Sprintf("%s:%s:", m.GroupId, m.ArtifactId))
+}
+
+func (m Module) SortText() string {
+	return fmt.Sprintf("%s:%s", m.GroupId, m.ArtifactId)
+}
+
+func (m Module) RepoName() api.RepoName {
+	return api.RepoName(fmt.Sprintf("maven/%s/%s", m.GroupId, m.ArtifactId))
+}
+
+func (m Module) CloneURL() string {
+	cloneURL := url.URL{Path: string(m.RepoName())}
+	return cloneURL.String()
+}
+
+type Dependency struct {
+	Module
+	Version         string
+	SemanticVersion *semver.Version
+}
+
+// SortDependencies sorts the dependencies by the semantic version in descending
+// order. The latest version of a dependency becomes the first element of the
+// slice
+func SortDependencies(dependencies []Dependency) {
+	sort.Slice(dependencies, func(i, j int) bool {
+		if dependencies[i].Module == dependencies[j].Module {
+			return dependencies[i].SemanticVersion.GreaterThan(dependencies[j].SemanticVersion)
+		}
+		return dependencies[i].Module.SortText() > dependencies[j].Module.SortText()
+	})
+}
+
+func (d Dependency) CoursierSyntax() string {
+	return fmt.Sprintf("%s:%s:%s", d.Module.GroupId, d.Module.ArtifactId, d.Version)
+}
+
+func (d Dependency) GitTagFromVersion() string {
+	return "v" + d.Version
+}
+
+func ParseMavenDependency(dependency string) Dependency {
+	parts := strings.Split(dependency, ":")
+	version := parts[2]
+	semanticVersion, _ := semver.NewVersion(version)
```

Same here, not sure if you should skip the error. I san invalid semver alright in a dependency struct based on its use in the rest of the program? If so it should be noted on the field. If not, we shouldn't allow construction of an illegal dependency.

---

I added a comment explaining why we ignore the error here
```

	// Ignore error from semantic version parsing because we only use the
	// semantic version for sorting dependencies, which falls back to
	// lexicographical ordering if the semantic version is missing. We can't
	// guarantee that every published Java package has a valid semantic
	// version according to the implementation of the Go-lang semver
	// package.
```